src/gui/qgsmaptip.cpp

```diff
@@ -313,6 +327,37 @@ QString QgsMapTip::fetchFeature( QgsMapLayer *layer, QgsPointXY &mapPosition, Qg
   return tipString;
 }
 
+QString QgsMapTip::fetchRaster( QgsMapLayer *layer, QgsPointXY &mapPosition, QgsMapCanvas *mapCanvas )
+{
+  QgsRasterLayer *rlayer = qobject_cast<QgsRasterLayer *>( layer );
+  if ( !rlayer )
+    return QString();
+
+  if ( !layer->isInScaleRange( mapCanvas->mapSettings().scale() ) )
+  {
+    return QString();
+  }
+
+  const QgsPointXY mappedPosition { mapCanvas->mapSettings().mapToLayerCoordinates( layer, mapPosition ) };
+
+  if ( ! layer->extent().contains( mappedPosition ) )
+  {
+    return QString( );
+  }
+
+  QString tipText { rlayer->mapTipTemplate() };
+
+  if ( ! tipText.isEmpty() )
+  {
+    QgsExpressionContext context( QgsExpressionContextUtils::globalProjectLayerScopes( layer ) );
+    context.appendScope( QgsExpressionContextUtils::mapSettingsScope( mapCanvas->mapSettings() ) );
+    context.appendScope( QgsExpressionContextUtils::mapLayerPositionScope( mappedPosition ) );
```

Comment:
Shouldn't we add also the mapLayerPositionScope into fetchfeature, for consistency?

Reply:
Yeah, good idea.